packages/mail-client/src/mail-client.ts

```diff
@@ -76,6 +76,12 @@ function getCredentials(
   destination: Destination,
   originalProperties: Record<string, any>
 ): { username: string; password: string } {
+  if (
+    destination.proxyType === 'OnPremise' &&
```

Comment:
Here, since internet smtp server can also have no username/password, I would leave this out.